packages/electrode-ignite/cli/ignite.js

```diff
@@ -0,0 +1,78 @@
+"use strict";
+
+const chalk = require("chalk");
+const xsh = require("xsh");
+
+const igniteCore = require("ignite-core");
+const pkg = require("../package.json");
+const logger = require("../lib/logger");
+const errorHandler = require("../lib/error-handler");
+let startTime;
+
+function checkElectrodeIgnite() {
+  xsh
+    .exec(true, "npm show electrode-ignite version")
+    .then(function(latestVersion) {
+      var latestVersion = latestVersion.stdout.slice(0, -1);
+
+      if (latestVersion > pkg.version) {
+        /* Case 1: electrode-ignite version outdated */
+        logger.log(
+          chalk.yellow(`Latest version is ${latestVersion} - trying to update.`)
+        );
+        xsh
+          .exec(true, `npm install -g electrode-ignite@${latestVersion}`)
+          .then(function() {
+            /* Auto update electrode-ignite version */
+            logger.log(
+              chalk.yellow(
+                `electrode-ignite updated to ${latestVersion}, exiting, please run your command again.`
+              )
+            );
+            return process.exit(0);
+          })
+          .catch(err =>
+            errorHandler(
+              err,
+              "Since it may not be safe for a module to update itself while running, " +
+                "please run the update command manually after electrode-ignite exits." +
+                `\nThe command is: npm install -g electrode-ignite@${latestVersion}`
+            )
+          );
+      } else if (latestVersion === pkg.version) {
+        /* Case 2: electrode-ignite latest version */
+        logger.log(
+          chalk.green("You've aleady installed the latest electrode-ignite.\n")
+        );
+
+        /* ignite-core */
+        igniteCore("oss", process.argv[2]);
+      } else {
+        /* Case 3: Invalid electrode-ignite version */
+        errorHandler(
+          "Failed at: Invalid electrode-ignite version. Please verify your package.json."
```

Comment:
this message is misleading.  If there's issue with electrode-ignite version, it's our problem, has nothing to do with user's package.json and nothing they can verify.  

First use proper semver comparison to check if version is newer, and then consider what error could occur and handle accordingly.